contracts/PBAB+Collabs/iyk/GenArt721CoreV2_IYK.sol

```diff
@@ -0,0 +1,195 @@
+     * @param _tokenName Name of token.
+     * @param _tokenSymbol Token symbol.
+     * @param _randomizerContract Randomizer contract.
+     * @param _startingProjectId The initial next project ID.
+     * @param _signVerifierRegistry Address of registry that resolves the signVerifierId.
+     * @param _signVerifierId ID of the signVerifier.
+     * @dev _startingProjectId should be set to a value much, much less than
+     * max(uint256) to avoid overflow when adding to it.
+     */
+    constructor(
+        string memory _tokenName,
+        string memory _tokenSymbol,
+        address _randomizerContract,
+        uint256 _startingProjectId,
+        address _signVerifierRegistry,
+        bytes32 _signVerifierId
+    )
+        GenArt721CoreV2_PBAB(
+            _tokenName,
+            _tokenSymbol,
+            _randomizerContract,
+            _startingProjectId
+        )
+    {
+        signVerifierRegistry = ISignVerifierRegistry(_signVerifierRegistry);
+        signVerifierId = _signVerifierId;
+    }
+
+    /**
+     * @notice Transfers a token from its owner to the recipient
+     * @param _sig The result of getClaimSigningHash signed by the signVerifier
+     * @param _blockExpiry The block as of which the signature is no longer valid
+     * @param _recipient The address that receives the token
+     * @param _tokenId The tokenId being claimed
+     * @dev ECDSA signatures are used to verify the permission to claim a NFT
+     */
+    function claimNFT(
+        bytes memory _sig,
+        uint256 _blockExpiry,
+        address _recipient,
+        uint256 _tokenId
+    ) public virtual {
+        bytes32 message = getClaimSigningHash(
+            _blockExpiry,
+            _recipient,
+            _tokenId
+        ).toEthSignedMessageHash();
+        address signVerifier = signVerifierRegistry.get(signVerifierId);
+        require(
+            ECDSA.recover(message, _sig) == signVerifier,
+            "Permission to call this function failed"
+        );
+        require(block.number < _blockExpiry, "Sig expired");
+
+        address from = ownerOf(_tokenId);
+        require(from != address(0));
+
+        claimNonces[_recipient]++;
+
+        _safeTransfer(from, _recipient, _tokenId, "");
+    }
+
+    /**
+     * @notice Returns the current claim nonce of a address
+     * @param _recipient The address whose nonce is being fetched
+     * @return nonce The addresses current nonce
+     * @dev This view exposes the nonce as it will be required for the signature.
+     * By including a nonce in the signature and updating the nonce on every claim,
+     * we prevent replay signature attacks, as signatures can only be used once.
+     */
+    function getClaimNonce(address _recipient)
+        external
+        view
+        virtual
+        returns (uint256)
+    {
+        return claimNonces[_recipient];
+    }
+
+    /**
+     * @notice Returns the hash that we expect was signed in claimNFT.
```

Comment:
Nit: I would encourage making it clear in the `@dev` note that the `claimNonces` value for the `_recipient` is also factored in here in the hash.